packages/node-resolve/src/index.js

```diff
@@ -100,6 +100,12 @@ export function nodeResolve(opts = {}) {
       // ignore IDs with null character, these belong to other plugins
       if (/\0/.test(importee)) return null;
 
+      if (/\0/.test(importer)) {
+        // handle cases like common-js plugin which has form
+        // \u0000<id>?commonjs-proxy
+        importer = importer.slice(1);
```

Comment:
not sure if there's a better way to handle this case?

If you remove this change you'll see the `pkg.browser with mapping to prevent bundle by specifying a value of false` test fail because the call to `resolveId` with the null byte is not allowed

Reply:
That one is a really good question that got overlooked so far. I do not agree with the solution, though. Files starting with `\0` are under the sole maintenance of some plugin, and we should not make assumptions about what the part behind the `\0` means.
I think a valid approach might be to treat the importer like `undefined`, which is the value that gets passed for entry points. At least for the commonjs plugin, this should work nicely as all imports from those proxy files are fully resolved absolute paths anyway, and I would expect the same from other plugins as well.

Reply:
>I think a valid approach might be to treat the importer like undefined, which is the value that gets passed for entry points

@lukastaegert I pushed a commit to do this